qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -297,7 +300,7 @@ def _recursively_handle_op(
 
         # Try to apply plugin mechanism
         decomposition = self._synthesize_op_using_plugins(op, qubits)
-        if decomposition:
+        if decomposition is not None:
             return decomposition, True
```

Comment:
This probably fixes a minor bug in the existing code, though previously after defining an empty circuit `qc = Quantum(3)`, the check `if qc:` would return `True`, but now it returns `False`.